knative-operator/pkg/common/openshift.go

```diff
@@ -20,6 +20,7 @@ func Mutate(ks *servingv1alpha1.KnativeServing, c client.Client) error {
 	stages := []func(*servingv1alpha1.KnativeServing, client.Client) error{
 		egress,
 		ingress,
+		configureIstio,
```

Comment:
Why is this needed for old CR sync'ing?

Reply:
The commit message describes it:
If we reconcile old -> new we override new's configMaps (because new's CR doesn't contain the istio config).

The istio config is currently added *after the fact* via a "manual" update to the ConfigMap itself and because `ApplyServiceMesh` is guarded by a state check it won't be retriggered after service mesh is ready once, causing us to loose that setting in the configMap.

I think all of the config map adjustions shall be done via the CR itself, not to the configmap directly, hence this change.